backend/src/web-server/controllers/history-query.controller.ts

```diff
@@ -174,13 +175,14 @@ export default class HistoryQueryController extends AbstractController {
         northManifest.settings
       );
 
+      const isCacheRestart = ctx.request.body.isCacheRestart as boolean;
       const itemsToAdd = ctx.request.body!.items.filter(item => !item.id);
       const itemsToUpdate = ctx.request.body!.items.filter(item => item.id);
       for (const itemId of ctx.request.body!.itemIdsToDelete) {
-        await ctx.app.reloadService.onDeleteHistoryItem(historyQuery.id, itemId);
+        await ctx.app.reloadService.onDeleteHistoryItem(historyQuery.id, itemId, isCacheRestart);
       }
-      await ctx.app.reloadService.onCreateOrUpdateHistoryQueryItems(historyQuery, itemsToAdd, itemsToUpdate);
-      await ctx.app.reloadService.onUpdateHistoryQuerySettings(ctx.params.id, command);
+      await ctx.app.reloadService.onCreateOrUpdateHistoryQueryItems(historyQuery, itemsToAdd, itemsToUpdate, isCacheRestart);
+      await ctx.app.reloadService.onUpdateHistoryQuerySettings(ctx.params.id, command, isCacheRestart);
```

Comment:
I suggest only the last one is set to `isCacheRestart`, the first two being set to false, to avoid calling three times the reset of the cache.
Also, I suggest to rename isCacheRestart into resetCache (same in the query param)